CMakeLists.txt

```diff
@@ -243,7 +243,20 @@ ENDIF()
 
 OPTION(WITH_MSAN "Enable memory sanitizer" OFF)
 IF (WITH_MSAN)
+  MY_CHECK_AND_SET_COMPILER_FLAG("-fdebug-macro" DEBUG)
   MY_CHECK_AND_SET_COMPILER_FLAG("-fsanitize=memory -fsanitize-memory-track-origins -U_FORTIFY_SOURCE" DEBUG RELWITHDEBINFO)
+  IF(NOT (have_C__fsanitize_memory__fsanitize_memory_track_origins__U_FORTIFY_SOURCE
+          AND have_CXX__fsanitize_memory__fsanitize_memory_track_origins__U_FORTIFY_SOURCE))
+    MESSAGE(FATAL_ERROR "Compiler doesn't support -fsanitize=memory flags")
+  ENDIF()
+  MY_CHECK_CXX_COMPILER_FLAG("-stdlib=libc++")
+  IF(have_CXX__stdlib_libc__)
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
+  ENDIF()
```

Comment:
I think that missing `-stdlib=libc++` must be treated as a fatal error. The server is built in C++ and it will depend on `libc++` or a similar library of the operating system. I would not attempt to support compiling other libraries (such as GCC `libstdc++`) with `clang -fsanitize=memory`.